app/service/file_svc.py

```diff
@@ -118,7 +120,18 @@ async def read_file(self, name, location='payloads'):
 
     def read_result_file(self, link_id, location='data/results'):
         buf = self._read(os.path.join(location, link_id))
-        return buf.decode('utf-8')
+        decoded_buf = buf.decode('utf-8')
+        try:  # Check if results file is valid dictionary
+            json.loads(self.decode_bytes(decoded_buf))
+            return decoded_buf
+        except json.JSONDecodeError:  # Exception occurs on non-dict result files (legacy)
```

Comment:
comment not needed

But what exactly do we mean by 'legacy' here?

Reply:
'Legacy' is referring to result files made prior to this change.  So these changes should still be able to read results from previously run operations (though it will place all output into 'stdout' of the dict, regardless of whether it was stdout or stderr).